examples/arm/executor_runner/arm_executor_runner.cpp

```diff
@@ -224,6 +224,65 @@ void et_pal_free(ET_UNUSED void* ptr) {}
 
 namespace {
 
+/// Lightweight heapless container that constructs and stores a T in-place.
```

Comment:
why not use std::optional? this appears roughly equivalent, but I read it quickly and might've missed something